pkg/util/provider/driver/fake.go

```diff
@@ -25,20 +25,31 @@ import (
 	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/status"
 )
 
+// VMs is the map to hold the VM data
+type VMs map[string]string
+
 // FakeDriver is a fake driver returned when none of the actual drivers match
 type FakeDriver struct {
 	VMExists       bool
 	ProviderID     string
 	NodeName       string
 	LastKnownState string
 	Err            error
+	fakeVMs        VMs
 }
 
 // NewFakeDriver returns a new fakedriver object
 func NewFakeDriver(fakeDriver *FakeDriver) Driver {
 	return fakeDriver
 }
 
+// AddMachine makes a call to the driver to create the machine.
+func (d *FakeDriver) AddMachine() error {
+	d.fakeVMs = make(VMs)
```

Comment:
This is incorrect. The array has to be initialized during construction of the struct instance.

Reply:
Understood!